tests/test_IJ2GUIWidget.py

```diff
@@ -182,112 +200,83 @@ def test_active_data_send(qtbot, ij, gui_widget: GUIWidget):
 @pytest.mark.skipif(
     running_headless(), reason="Only applies when not running headlessly"
 )
-def test_active_data_receive(qtbot, ij, gui_widget: GUIWidget):
+def test_active_data_receive(asserter, qtbot, ij, gui_widget: GUIWidget):
     button: FromIJButton = gui_widget.from_ij
     assert not button.isEnabled()
 
     # Show the button
     qtbot.mouseClick(gui_widget.gui_button, Qt.LeftButton, delay=1)
-    qtbot.waitUntil(lambda: button.isEnabled())
+    asserter(lambda: button.isEnabled())
 
     # Add some data to ImageJ
-    sample_data = jc.ArrayImgs.bytes(10, 10, 10)
+    sample_data = jc.ArrayImgs.longs(10, 10, 10)
     ij.ui().show("test_from", sample_data)
+    asserter(lambda: ij.display().getActiveDisplay() is not None)
 
     # Press the button, handle the Dialog
     assert 0 == len(button.viewer.layers)
     qtbot.mouseClick(button, Qt.LeftButton, delay=1)
 
     # Assert that the data is now in napari
-    assert 1 == len(button.viewer.layers)
+    asserter(lambda: 1 == len(button.viewer.layers))
     layer = button.viewer.layers[0]
     assert isinstance(layer, Image)
     assert (10, 10, 10) == layer.data.shape
 
 
-@pytest.mark.skipif(
-    running_headless(), reason="Only applies when not running headlessly"
-)
-def test_chosen_data_send(qtbot, ij, gui_widget_chooser):
-    button: ToIJButton = gui_widget_chooser.to_ij
-    assert not button.isEnabled()
-
-    # Show the button
-    qtbot.mouseClick(gui_widget_chooser.gui_button, Qt.LeftButton, delay=1)
-    qtbot.waitUntil(lambda: button.isEnabled())
-
-    # Add some data to the viewer
-    sample_data = numpy.ones((100, 100, 3))
-    image: Image = Image(data=sample_data, name="test_to")
-    current_viewer().add_layer(image)
-
-    handler = _HandleChecker()
-
-    def handle_dialog(handler: _HandleChecker):
-        qtbot.waitUntil(lambda: isinstance(QApplication.activeWindow(), QDialog))
+def _handle_button_dialog(asserter, qtbot, button):
+    def handle_dialog():
+        asserter(lambda: isinstance(QApplication.activeWindow(), QDialog))
         dialog = QApplication.activeWindow()
         buttons = dialog.findChildren(QPushButton)
         assert 2 == len(buttons)
         for button in buttons:
             if "ok" in button.text().lower():
-                qtbot.mouseClick(button, Qt.LeftButton, delay=1)
-                handler.done_handling()
+                button.clicked.emit()
                 return
         pytest.fail("Could not find the Ok button!")
 
```

Comment:
Can we just return `false` here? I don't know that this error is being propagated to the end.

Reply:
This code was removed, so no longer necessary